phase1-coordinator/src/objects/round.rs

```diff
@@ -816,7 +830,8 @@ impl Round {
                             true,
                         )
                     };
-                    if storage.exists(&contribution_file) {
+                    // Don't remove initial challenge
+                    if storage.exists(&contribution_file) && chunk.current_contribution()?.get_contributor().is_some() {
```

Comment:
I have some doubts about this line. @howardwu what do you think about this change? There's a case when initial challenge gets removed when we drop a contributor, but it doesn't get created later on. I wanted to make a check here: if the challenge is initial, then don't remove it.

Reply:
Now I'm not sure about the "initial challenge" terminology. Do we produce all of the challenges for all of the contributions during the round initialization? Probably not, and probably some new challenges are created after the tasks are completed and verified. This part of code is quite unclear to me.

But thanks to your comment, I had some new ideas and was able to find one more bug in the drop logic - #243

Reply:
challenge is what gets sent to the contributor each time it performs a task... I think? This differs from the contributions that are present at the start of the round, which have no contributor, and are created by the verifier right? :sweat_smile:

Reply:
If you're unsure, revert it. This line was from the PR merged 7 days ago.

Reply:
@howardwu can you elaborate a bit more on this?

> This line was from the PR merged 7 days ago.

We havent merged any PR's changing this line in the last few weeks, if I recall corectly. Am I missing something?

The part `storage.remove(&contribution_file)` did exist in the code a few months ago. Then we had an error `Locator in call to remove() doesn't exist in storage.` and I changed this line to
```rust
if storage.exists(&contribution_file) {
    storage.remove(&contribution_file)?;
}
```
The part I'm unsure of is the overall block which starts with
```rust
// Remove the next challenge locator if the current response has been verified.
if chunk.current_contribution()?.is_verified() {
```
The error we are having is produced by the following:
- first the round initialization is called
- then the task is locked
- then contributor is dropped

During the drop logic the code block I've mentioned above removes "the next challenge locator", wchich is in fact the challenge generated during the round initialization. And if we remove this particular challenge, it will not be created later on, leading to an error when the replacement contributor tries to contribute.

One more problem is that the challenge created during initialization contains `verified: true`, that's why the code inside this `if` block is executed.

Reply:
@kellpossible I'm not sure if I can answer your question, but if we start the coordinator and check `round1/manifest.json` there will be lines like
```json
"open": [
    "./transcript/development/coordinator.json",
    "./transcript/development/round_0/chunk_0/contribution_0.verified",
    "./transcript/development/round_0/chunk_1/contribution_0.verified",
```
It feels like the `chunk_0/contribution_0.verified` file and similar files are the ones which can sometimes be removed during the drop contributor logic

Reply:
If we decided to make this a TODO item, it would be good to create a corresponding github issue, and reference the issue in the TODO so we can keep track of it.